assets/js/leaves.js

```diff
@@ -208,14 +221,21 @@ class Leaf {
         }
 
         panel.querySelector("article").replaceWith(article);
-        // scroll to the top, in case previous leaf was scrolled
-        panel.scrollTop = 0;
-        // make sure panel is active
-        panel.parentElement.classList.add("show-details");
-        panel.parentElement.classList.remove("closed");
         // store current leaf url in a data attribute so we can check for reload
         panel.dataset.showing = leafTarget.dataset.url;
+      })
+      .catch((response) => {
+        // clone error article and pass in to article
+        let errorArticle = document.querySelector("#leaferror").cloneNode(true);
+        panel.querySelector("article").replaceWith(errorArticle);
+        errorArticle.classList.add("showing");
       });
+
+    // scroll to the top, in case previous leaf was scrolled
+    panel.scrollTop = 0;
+    // make sure panel is active
+    panel.parentElement.classList.add("show-details");
+    panel.parentElement.classList.remove("closed");
```

Comment:
I think these are redundant now, with my latest panel code changes in develop

Reply:
Do you mean we only need one of these, or neither, once the new changes from develop are in? Looks like I need only the first of these lines to create the behavior I want. Should I just merge develop into my PR now?

Reply:
I think you shouldn't need them at all — opening and closing the panel should be handled in the panel code, leaf code shouldn't have to know about the container (ideally! but possibly not practically / not yet).

Yes, probably a good idea to update from develop and merge your changes with mine from the other branch (hopefully not to hard to resolve any conflicts)